stackdriver.go

```diff
@@ -439,6 +441,12 @@ func (e *Exporter) StopMetricsExporter() {
 func (e *Exporter) Close() error {
 	tErr := e.traceExporter.close()
 	mErr := e.statsExporter.close()
+	// If the trace and stats exporter share client connections,
```

Comment:
This looks like a reasonable local fix, but should the client connection be tracked better? E.g. should there be a reference count?

Reply:
In practice, this case shouldn't ever really happen, since you normally don't want to dial the same endpoint for metrics and traces.  It just happens in unit tests for the collector, from what I can tell.  If we did something like that, we would basically just want to only close one connection if the user provides the WithGRPCConn option, since they connections are different otherwise.

But it would have been a bit ugly to check and track the args that are currently just passed through to the monitoring and trace clients.  The other option would've been to add a "metrics-only" exporter constructor function, but that also seemed like too much for fixing unit tests.

The other option would be for whoever creates the connection initially to be responsible for closing it.  But in most cases, users don't create the connection themselves, and will still need to invoke Shutdown (in the collector) for other reasons.